platform/darwin/src/NSDate+MGLAdditions.mm

```diff
@@ -1,15 +1,14 @@
 #import "NSDate+MGLAdditions.h"
 
-@implementation NSDate (MGLAdditions)
-
-mbgl::Duration MGLDurationInSecondsFromTimeInterval(NSTimeInterval duration)
+mbgl::Duration MGLDurationFromTimeInterval(NSTimeInterval duration)
 {
     return std::chrono::duration_cast<mbgl::Duration>(std::chrono::duration<NSTimeInterval>(duration));
 }
 
-NSTimeInterval MGLTimeIntervalFromDurationInSeconds(mbgl::Duration duration)
+NSTimeInterval MGLTimeIntervalFromDuration(mbgl::Duration duration)
 {
-    return duration.count();
+    std::chrono::nanoseconds nano(duration.count());
```

Comment:
This implementation makes the unsafe assumption that `mbgl::Duration`'s count is in nanoseconds. `mbgl::Duration` is a typedef to [`std::chrono::steady_clock::duration`](http://en.cppreference.com/w/cpp/chrono/steady_clock), and `std::chrono::steady_clock` does not specify a particular tick period.

In addition, this implementation truncates any fractional seconds in the duration. `NSTimeInterval` is a floating-point type, and so the conversion should be expected to preserve fractional seconds.

I believe the correct conversion is as follows (but don't trust me -- write some more tests 😄):

```
NSTimeInterval MGLTimeIntervalFromDuration(mbgl::Duration duration)
{
    std::chrono::duration<NSTimeInterval, std::ratio<1>>(duration).count();
}
```